server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/MailboxGetMethod.scala

```diff
@@ -77,18 +78,22 @@ class MailboxGetMethod @Inject() (serializer: Serializer,
 
   private def getMailboxes(mailboxGetRequest: MailboxGetRequest, mailboxSession: MailboxSession): SFlux[MailboxGetResults] = mailboxGetRequest.ids match {
     case None => getAllMailboxes(mailboxSession).map(MailboxGetResults.found)
-    case Some(ids) =>
-      getAllMailboxes(mailboxSession)
-        .collectSeq()
-        .flatMapMany(mailboxes => SFlux.merge(Seq(
-          SFlux.fromIterable(mailboxes)
-            .filter(mailbox => ids.value.contains(mailbox.id))
-            .map(MailboxGetResults.found),
-          SFlux.fromIterable(ids.value)
-            .filter(id => !mailboxes.map(_.id).contains(id))
-            .map(MailboxGetResults.notFound))))
+    case Some(ids) => SFlux.fromIterable(ids.value)
+      .flatMap(id => getMailboxByIdOrThrow(id, mailboxSession))
   }
 
+  private def getMailboxByIdOrThrow(mailboxId: MailboxId, mailboxSession: MailboxSession): SMono[MailboxGetResults] =
+    quotaFactory.loadFor(mailboxSession)
+      .subscribeOn(Schedulers.elastic)
```

Comment:
Should the subscribe be at the end of the chain ?

Reply:
If I do this I get back the error I got before when compiling: https://github.com/linagora/james-project/pull/3400#discussion_r445296137 (this one being a mystery to me)

So...